Dockerfile-multi-arch

```diff
@@ -0,0 +1,50 @@
+ARG svc_name
+
+# go build stage
+FROM --platform=$BUILDPLATFORM golang:1.15.0 as golang
+WORKDIR /emojivoto-build
+
+# install protobuf
+RUN apt-get update && apt-get install -y protobuf-compiler
+RUN go get github.com/golang/protobuf/protoc-gen-go
+
+# cache go dependencies
+COPY go.mod go.sum .
+RUN go mod download
+
+# compile
+COPY . .
+ARG TARGETARCH
+ARG svc_name
+RUN GOARCH=$TARGETARCH make -C $svc_name clean protoc compile
+
+# webpack stage
+FROM --platform=$BUILDPLATFORM node:14-buster as webpack-bundle
+WORKDIR /emojivoto-build
+COPY . .
+RUN make -C emojivoto-web clean webpack package-web
+
+FROM golang as build-emojivoto-emoji-svc
+FROM golang as build-emojivoto-voting-svc
+FROM golang as build-emojivoto-web
```

Comment:
yes. this is the real definition of each svc.
the `FROM build-$svc_name ` is just a placeholder, when it is on runtime, it will be compiled to the real svc name, for example to `emojivoto-web ` then it will use the `FROM golang as build-emojivoto-web `.
Kind of pattern matching style